executor/analyze.go

```diff
@@ -135,6 +161,34 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if err != nil {
 		return err
 	}
+	if needGlobalStats {
+		for _, info := range globalStatsMap {
+			globalStats, succ := statsHandle.MergePartitionStats2GlobalStats(infoschema.GetInfoSchema(e.ctx), info.tableID, info.isIndex, info.idxID)
+			if succ {
+				for i := 0; i < globalStats.Num; i++ {
+					hg := globalStats.Hg[i]
+					cms := globalStats.Cms[i]
+					topN := globalStats.TopN[i]
+					if hg == nil || cms == nil || topN == nil {
```

Comment:
Does nil appear in the above three variables, but they are correct or expected?

Reply:
This check seems weird and unnecessary. `MergePartitionStats2GlobalStats ` should ensure that when `succ` is true, all elements in `globalStats` are not nil.